clouddrift/ragged.py

```diff
@@ -854,6 +854,58 @@ def unpack(
     return [unpacked[i] for i in rows]
 
 
+def index_to_row(
+    index: int | list[int] | np.ndarray,
+    rowsize: list[int] | np.ndarray | xr.DataArray,
+) -> list:
+    """Obtain a list of row indices from a list of observation indices of a ragged array.
+
+    Parameters
+    ----------
+    index : int or list or np.ndarray
+        A integer observation index or a list of observation indices of a ragged array.
+    rowsize : list or np.ndarray or xr.DataArray
+        A sequence of row sizes of a ragged array.
+
+    Returns
+    -------
+    list
+        A list of row indices.
+
+    Examples
+    --------
+    To obtain the row index of observation 5 within a ragged array of three consecutive
+    rows of sizes 2, 4, and 3:
+    >>> index_to_row(5, [2, 4, 3])
+    [1]
+
+    To obtain the row indices of observations 0, 2, and 4 within a ragged array of three consecutive
+    rows of sizes 2, 4, and 3:
+    >>> index_to_row([0, 2, 4], [2, 4, 3])
+    [0, 1, 1]
+
+    """
+    # if index is an integer, convert it to a list
+    if isinstance(index, int) | isinstance(index, float):
```

Comment:
We likely want this check to only appy to values that are of type int. The right side of the or operator should be removed.

Reply:
can we allow to pass a float pretending to be an integer?

Reply:
not sure I follow the question, is this regarding the static type checking or whether we can pass a float like 3.18 but instead interpret it as the integer 318?

Reply:
pass 5.0 for 5

Reply:
I don't think we should allow the use of floats if the intent of the function is to find the row_idx from the obs_idx where we know obs_idx is an integer. 

If the intent is instead to pass in a data var say like: `sst` and find the associated row_idx of the sst observation (determining the sst observations obs_index first but encapsulating the logic) then it makes sense to support this type but we would also need to update the logic.

I think my main confusion here is figuring out the case where an obs_idx should be a float rather than an integer.

Reply:
I am not sure, but I would easily see myself generating a vector of indices which contains float instead of integer